api/common.js

```diff
@@ -413,6 +416,125 @@ exports.load_github_detail = function(service_name, cb) {
+    let response_mag = ""
+    axios({
+        method: 'get',
+        url: url_mag,
+        responseType: 'json',
+        headers: headers
+      }).then(function (response) {
+        //   console.log(response)
+        // console.log(response.data)
+        this.response_mag = response
+        if(response.data.entities.length > 0){
+            // db.Projects.findByIdAndUpdate(projectid, {$set: {"mag": response.data.entities}}, {new: true});
+            data_mag = response.data.entities
+            // console.log(data_mag[0])
+            var mag_resp;
+            for (i=0; i<data_mag.length;i++){
+                var object = data_mag[i];   // get first object
+
+                var date = Date.parse(object.D);      // get Date and 
+                object.publicationDate = new Date(date); // assign to new key
+                
+                var citationCount = object.CC; //get CC
+                // console.log(citationCount)
+                object.citationCount = citationCount; // add Citation Count
+                
+                var title = object.Ti; //get Ti
+                object.title = title; // add Title
+
+                var venue = object.VFN; //get VFN
+                object.venue = venue; // add journal
+                
+                for (var y = 0; y < object.AA.length; y++){
+                    var name = object.AA[y].DAuN;
+                    object.AA[y].name = name;
+                    object.AA[y].institution = object.AA[y].AfN;
+
+                    delete object.AA[y].DAuN;
+                    delete object.AA[y].AuId;
+                    delete object.AA[y].AfId;
+                    delete object.AA[y].AfN;
+                    delete object.AA[y].AuN;
+                    // var institution = object.AA[y].AfN;
+                    // authors[y].name = name;
+                    // authors[y].institution = institution;
+                }
+
+                object.authors = object.AA;
+                object.fields = []
+                for (var y = 0; y < object.F.length; y++){
+                        // object.fields += ","+object.F[y].FN;
+                        object.fields.push(object.F[y].FN)
+                }
+
+                if(typeof object.IA!== "undefined"){
+                    object.abstract = "";
+                try {
+                    var ia_obj = JSON.stringify(object.IA);
+                    ia_obj2 = JSON.parse(ia_obj);
+                    var keys = Object.keys(ia_obj2.InvertedIndex);
+                    Object.keys(ia_obj2.InvertedIndex).forEach(function (item) {
+                        object.abstract += item+" ";
+                    });
+                    delete object.IA
+                } catch (error) {
+                    // console.log(error)
+                }
+            }
+
+                delete object.VFN;
```

Comment:
Instead of deleting things you don't want, you should probably copy/set things you want?

Reply:
It would create a duplicate copy with another name then. Do we want to store the original data too?

Reply:
Yeah, I think you should start with an empty object and set things as necessary. If Microsoft adds extra things in the future, we don't want that to suddenly be stored in our DB.

> Do we want to store the original data too?
I don't think so.. if we need the original data, we can just make a query again using the same `Id` right?

Reply:
Yes by using the query Id=paperID